### PR TITLE
Implement Set#intersect?

### PR DESCRIPTION
**Pull request by yui-knk**

### CODE REVIEW

File: spec/std/set_spec.cr [hunk 1]
```diff
@@ -74,6 +74,24 @@ describe "Set" do
...
+
+    set.intersect?(set).should be_true
+    set.intersect?(Set{2, 4}).should be_true
+    set.intersect?(Set{5, 6, 7}).should be_true
+    set.intersect?(Set{1, 2, 6, 8, 4}).should be_true
+
+    set.intersect?(empty_set).should be_false
+    set.intersect?(Set{0, 2}).should be_false
+    set.intersect?(Set{0, 2, 6}).should be_false
+    set.intersect?(Set{0, 2, 6, 8, 10}).should be_false
```
Comment (PragTob) [inline spec/std/set_spec.cr]: Imo the different checks/assertions should be split up into multiple test cases - makes it easier to identify what broke if something broke :)

**Comment by asterite** [comment]
Following the naming convention we have, I would call it `interesects?`. For example `if this_set.interesects?(the_other_set)` is correct English, but with `interesect?` it is not.

**Comment by yui-knk** [comment]
@asterite  Updated

**Comment by asterite** [comment]
Thanks!

**Pull request merged**